### PR TITLE
441/fix/clear cell while opening it by typing and delete value when I hit delete / backspace.

### LINKED ISSUE #441: AAU, when opening cell by typing text, I want to have cell value cleared before my text is entered.
By default, when we are opening a table cell by typing text, it should erase before entering the keys we just typed in. 

Because those values are controlled by different inputs, and edit mode is a generic implementation, we have to find a way to have maybe a resetValue hook that has to be implemented in any editable cell, or maybe implement this behavior directly into the editable cell editable component ?

### PR DESCRIPTION
**Pull request by abhithory**
Fixed #441  and #447 

- When you open any cell (except that redirects to another page) by typing then first it will clear that and then type
- Also it will delete that cell when you hit delete / backspace

### CODE REVIEW

**Comment by abhithory** [comment]
@lucasbordeau @FelixMalfait , please review this and let me know if you want any improvment

File: front/src/modules/ui/field/hooks/useResetField.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import { isFieldPhone } from '../types/guards/isFieldPhone';
+import { isFieldProbability } from '../types/guards/isFieldProbability';
+import { isFieldRelation } from '../types/guards/isFieldRelation';
+import { isFieldText } from '../types/guards/isFieldText';
+import { isFieldURL } from '../types/guards/isFieldURL';
+
+import { useIsFieldEmpty } from './useIsFieldEmpty';
+import { usePersistField } from './usePersistField';
+
+export const useResetField = () => {
```
Comment (lucasbordeau) [inline front/src/modules/ui/field/hooks/useResetField.ts:23]: This should be scoped to the table, so to be moved in data table folder and renamed like `useResetTableCell`

File: front/src/modules/ui/field/hooks/useResetField.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  const resetField = () => {
+    if (
+      (isFirstColumnCell && fieldDefinition.basePathToShowPage) ||
+      isFieldEmpty
+    ) {
+      return;
+    }
+
+    const resetValue = isFieldRelation(fieldDefinition)
```
Comment (lucasbordeau) [inline front/src/modules/ui/field/hooks/useResetField.ts:39]: We can abstract this in a hook or util `getFieldResetValue(fieldDefinition)`

File: front/src/modules/ui/field/hooks/useResetField.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      ? null
+      : isFieldText(fieldDefinition)
+      ? ''
+      : isFieldEmail(fieldDefinition)
+      ? ''
+      : isFieldDate(fieldDefinition)
+      ? 'return'
```
Comment (lucasbordeau) [inline front/src/modules/ui/field/hooks/useResetField.ts:46]: This is not a value and shouldn't be in the same function.

We shouldn't mix both behaviors : 
- Getting the reset value
- Checking if it's a date so we apply a special case

File: front/src/modules/ui/field/hooks/useResetField.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      : isFieldChip(fieldDefinition)
+      ? ''
+      : isFieldDoubleTextChip(fieldDefinition)
+      ? { firstValue: '', secondValue: '' }
+      : isFieldDoubleText(fieldDefinition)
+      ? { firstValue: '', secondValue: '' }
+      : isFieldMoney(fieldDefinition)
+      ? null
+      : '';
+    if (resetValue === 'return') {
```
Comment (lucasbordeau) [inline front/src/modules/ui/field/hooks/useResetField.ts:66]: We should check if it's a date directly instead.

**Review by lucasbordeau — CHANGES_REQUESTED** [review]
A few important concepts to pay attention to when working on this tricky part of the app : 
- Erasing the value should be done only in the context of the component's edit mode state, persisting is a feature that should be left to when we're closing the cell, not when we're opening it.
- The implementation you've made is for the field logic including InlineCell, we want it only for TableCell, so it should be put in `data-table/table-cell` module

You'll get it but don't worry if it takes a few tries, it's a complex part of the app.

File: front/src/modules/ui/field/hooks/useResetField.ts (hunk 3 above)
Comment (abhithory) [reply front/src/modules/ui/field/hooks/useResetField.ts:46]: yes i am doing that. i am returning that and not setting anything if its data.

what should i set when its data?

File: front/src/modules/ui/field/hooks/useResetField.ts (hunk 4 above)
Comment (abhithory) [reply front/src/modules/ui/field/hooks/useResetField.ts:66]: ok, then what should we set?

**Comment by abhithory** [comment]
Ok thanks @lucasbordeau , i will do these things

File: front/src/modules/ui/field/hooks/useResetField.ts (hunk 2 above)
Comment (abhithory) [reply front/src/modules/ui/field/hooks/useResetField.ts:39]: should i create a new hook for this?? or just create new function in this `useResetTableCell` ??
i think this we can create a new Function `getResetValue` and then we can use that in `resetField` function. also we can export this `getResetValue` function from this hook.

What do you think?

**Comment by abhithory** [comment]
Hi @lucasbordeau ,
I am done with other changes but i an still confused in this thing little bit. 
Can you please help me in this?

You said `Erasing the value should be done only in the context of the component's edit mode state, persisting is a feature that should be left to when we're closing the cell, not when we're opening it.`

when we are opening the cell by typing anything on it, we should not use `persistField`. Right?

So, now what should i use here to clear only the internal value of the input?? 
- should i create a recoil state and set its reset value?
- should i add `ref` to the input and then set its reset value??
- or is there any other way to do this??


also, while deleting the cell with Delete/Backspace button can we use persistField or is there anything else for this???

**Comment by lucasbordeau** [comment]
@abhithory As discussed on Discord, the recommendations : 
- Don't use refs
- Add a new atom for openingAction (idea for the type : `type OpeningAction = 'enter' | 'clic' | 'backspace' | 'text-input'`)
- Use this new atom to change the initial value of the different low-level inputs.

**Review by lucasbordeau — APPROVED** [review]

**Pull request merged**